### PR TITLE
Enable/Disable authentication maps

### PR DESCRIPTION
**Pull request by dmzoneill**

### CODE REVIEW

File: ansible_base/authentication/utils/claims.py [hunk 1]
```diff
@@ -62,9 +62,13 @@ def create_claims(authenticator: Authenticator, username: str, attrs: dict, grou
         trigger_result = TriggerResult.SKIP
         allowed_keys = TRIGGER_DEFINITION.keys()
         invalid_keys = set(auth_map.triggers.keys()) - set(allowed_keys)
+
+        if auth_map.enabled is False:
+            logger.warning(f"The AuthenticatorMap {auth_map.id} is disabled")
```
Comment (relrod) [inline ansible_base/authentication/utils/claims.py]: Maybe info instead? I feel like we're too quick to use warning sometimes - we should save that for things that aren't normal operating conditions. If we're supporting disabling authenticator maps, then that's a normal state for them to be in and warning about it seems excessive.

File: ansible_base/authentication/models/authenticator_map.py [hunk 2]
```diff
@@ -88,3 +88,8 @@ class Meta:
             )
         ),
     )
+    enabled = models.BooleanField(
+        null=False,
+        default=False,
```
Comment (relrod) [inline ansible_base/authentication/models/authenticator_map.py]: `False` here but `True` in the migration - is why CI is failing.

File: ansible_base/authentication/models/authenticator_map.py [hunk 3]
```diff
@@ -88,3 +88,8 @@ class Meta:
             )
         ),
     )
+    enabled = models.BooleanField(
+        null=False,
+        default=True,
+        help_text=_("Enables or disables this authentication map"),
```
Comment (relrod) [inline ansible_base/authentication/models/authenticator_map.py]:
```suggestion
        help_text=_("Enables or disables this authenticator map"),
```

File: test_app/tests/authentication/utils/test_claims.py
Comment (relrod) [inline test_app/tests/authentication/utils/test_claims.py:1]: Can we have a test for what happens when it's disabled? Just something that checks that it doesn't get applied.

**Review by jay-steurer — APPROVED** [review]

**Review by john-westcott-iv — APPROVED** [review]

**Pull request merged**